### PR TITLE
parametrize the preset json file name in the spring boot configuration

### LINKED ISSUE #10553: preset.json in spring config
See https://github.com/jhipster/jhipster-lite/blob/main/src/main/java/tech/jhipster/lite/project/infrastructure/secondary/FileSystemProjectsRepository.java#L29

Instead of hard coding this file, it should be nice to allow this as config in spring properties

Several use cases:
- in jhlite extension, we could use specific presets
- in spring profile vue, we could have vue instead of angular
- etc

### PR DESCRIPTION
**Pull request by renanfranca**
Fix #10553

### CODE REVIEW

**Comment by murdos** [comment]
What exactly has changed since https://github.com/jhipster/jhipster-lite/pull/10578?

**Comment by renanfranca** [comment]
> What exactly has changed since https://github.com/jhipster/jhipster-lite/pull/10578?

I tagged you in the code comments with explanations. Let me know if it was enough 👍

**Comment by murdos** [comment]
> I tagged you in the code comments with explanations. Let me know if it was enough 👍

I have not received any notification. Maybe you've started a review session that you have not finished?

**Comment by renanfranca** [comment]
> I have not received any notification. Maybe you've started a review session that you have not finished?


@murdos: Maybe I don't know how to start a review properly when I am the PR owner. I'll play around with two GitHub accounts in another repository to master it, hahaha ;)

### Changes since the previous PR #10578

The bug was fixed by adding this parameter:

```java
@Configuration 
@EnableConfigurationProperties(JHipsterPresetFileProperties.class) 
class JHipsterPresetFilePropertiesConfiguration { 
  
  @Bean 
  JHipsterPresetFile jhipsterPresetFile(JHipsterPresetFileProperties presetFileProperties) { 
    return new JHipsterPresetFile(PresetName.from(presetFileProperties.getName())); 
  } 
}
```

This was the previous bugged code:

```java
@Configuration 
@EnableConfigurationProperties(JHipsterPresetFileProperties.class) 
class JHipsterPresetFilePropertiesConfiguration { 
  
  @Bean 
  JHipsterPresetFile jhipsterPresetFile() { 
    return new JHipsterPresetFile(new JHipsterPresetFileProperties().getName()); 
  } 
}
```

I used a value object instead of a string and refactored the code to fit that change:

```java
package tech.jhipster.lite.project.domain.resource; 
  
import tech.jhipster.lite.project.domain.preset.PresetName; 
import tech.jhipster.lite.shared.error.domain.Assert; 
  
public record JHipsterPresetFile(PresetName name) { 
  public JHipsterPresetFile { 
    Assert.notNull("name", name); 
  } 
}
```

**Review by murdos — APPROVED** [review]

File: src/main/java/tech/jhipster/lite/project/infrastructure/secondary/JHipsterPresetFilePropertiesConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.context.annotation.Configuration;
+import tech.jhipster.lite.project.domain.preset.PresetName;
+import tech.jhipster.lite.project.domain.resource.JHipsterPresetFile;
+
+@Configuration
+@EnableConfigurationProperties(JHipsterPresetFileProperties.class)
+class JHipsterPresetFilePropertiesConfiguration {
+
+  @Bean
+  JHipsterPresetFile jhipsterPresetFile(JHipsterPresetFileProperties presetFileProperties) {
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/project/infrastructure/secondary/JHipsterPresetFilePropertiesConfiguration.java:14]: in fact, I would use directly @Value in repository, to inject the property, no need to use a Configuration and a Bean

**Comment by renanfranca** [comment]
https://github.com/jhipster/jhipster-lite/pull/10590#discussion_r1721449763

@murdos , please WDYT about it? (I tagged here, because I am not sure if you received notification 😅)

**Pull request merged**